src/enkf/loadbal.f90

```diff
@@ -460,52 +511,94 @@ subroutine gather_chunks
 ! scounts is number of data elements to send to processor np.
 ! rcounts is number of data elements to recv from processor np.
 ! displs is displacement into send array for data to go to proc np
-if (nproc <= ntasks_io-1) then
-   rcounts = nanals_per_iotask*npts_max*ncdim
-else
-   rcounts = 0
-endif
-do np=0,numproc-1
-   displs(np) = np*nanals_per_iotask*npts_max*ncdim
-   if (np <= ntasks_io-1) then
-      scounts(np) = nanals_per_iotask*npts_max*ncdim
-   else
-      scounts(np) = 0
-   end if
-enddo
-allocate(recvbuf(numproc*nanals_per_iotask*npts_max*ncdim))
-allocate(sendbuf(numproc*nanals_per_iotask*npts_max*ncdim))
-
 
-do nb=1,nbackgrounds ! loop over time levels in background
-  do nn=1,ncdim
-    do i=1,numptsperproc(nproc+1)
-      do nanal=1,nanals
-         n = ((nanal-1)*ncdim + (nn-1))*npts_max + i
-         ! add ensemble mean back in.
-         sendbuf(n) = anal_chunk(nanal,i,nn,nb)+ensmean_chunk(i,nn,nb)
-         ! convert to increment (A-F).
-         sendbuf(n) = sendbuf(n)-(anal_chunk_prior(nanal,i,nn,nb)+ensmean_chunk_prior(i,nn,nb))
-      enddo
-    enddo
-  enddo
-  call mpi_alltoallv(sendbuf, scounts, displs, mpi_real4, recvbuf, rcounts, displs,&
-                     mpi_real4, mpi_comm_world, ierr)
-  if (nproc <= ntasks_io-1) then
-    do np=1,numproc
-     do ne=1,nanals_per_iotask
-      do nn=1,ncdim
-       do i=1,numptsperproc(np)
-         n = ((np-1)*ncdim*nanals_per_iotask + (ne-1)*ncdim + (nn-1))*npts_max + i
-         grdin(indxproc(np,i),nn,nb,ne) = recvbuf(n)
+if (real(numproc)*real(nanals_per_iotask)*real(npts_max)*real(ncdim) < 2**32/2. - 1) then
```

Comment:
Same comment as for `controlvec.f90`.   One option is to replace `1` and `2` with `one` and `two` from the constants module.   That is, `2**32/2. - 1` becomes `two**32/two - one`.

Reply:
Ignore my bad suggestion.  `two**32/two -one` does not work.  Using this expression alters the analyses. 
  `2_i_kind**32/2._real - 1_i_kind` works.  This expression yields analysis files that are bitwise identical with the original code.

Reply:
@RussTreadon-NOAA Thanks for investigation on it.  It is interesting, why "two**32/two-one"  alters the analyses. It reminds me that my verification for fv3-reg doesn't test the new scheme ( only invoked the "old" procedure decided by evaluation here using  (2**31-1) )  . I will rerun a verification by forcing use of the new procedure brought up by this PR .  I will finish it not later than tomorrow .